Sources/NIOKit/FutureExtensions.swift

```diff
@@ -0,0 +1,17 @@
+// MARK: - Guard
```

Comment:
If you put this inside of the extension, it should generate a nice header in our API docs